### DISCUSSION TITLE
How to bump patch version on each commit to main?

### DISCUSSION THREAD

**Discussion by jameswoodley** [GitTools/GitVersion#4341] [Q&A] [answered]
We are using Octopus Deploy, and need to have each push to main bump the patch version.

We tag on major and minor versions, so that part is as expected, but we just want each commit on main to bump the patch. Instead it is keeping the patch as-is, but adding a pre-release version. I.e. 9.0.1-4

We need in this instance each push to main to make a 9.0.2, 9.0.3, 9.0.4 and so on.

Currently we are using 
```
assembly-versioning-scheme: MajorMinorPatch
mode: ContinuousDelivery
```

What else do we need to do to lose the pre-release suffix, and just bump the patch on each release into Octopus?

**Comment by HHobeck** [comment, marked as answer]
For this you need to use the mainline version strategy which is included in the `TrunkBased/preview1` workflow:

```yaml
workflow: TrunkBased/preview1
```

Link[1]:
https://gitversion.net/docs/reference/configuration#snippet-/docs/workflows/TrunkBased/preview1.yml
